### PR TITLE
Implemented makeCluster and save functions to reduce boilerplate in test code

### PR DESCRIPTION
**Pull request by akarukappadath**
This is a preliminary PR that contains a few test file updates that correspond to the new functions. 

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by akarukappadath** [comment]
jenkins retest

File: src/test/scala/org/broadinstitute/dsde/workbench/leonardo/CommonTestData.scala [hunk 1]
```diff
@@ -12,18 +12,19 @@ import org.broadinstitute.dsde.workbench.leonardo.auth.sam.MockPetClusterService
...
 import org.scalatest.concurrent.ScalaFutures
 
 import scala.concurrent.ExecutionContext
-import scala.concurrent.duration._
+
 
 // values common to multiple tests, to reduce boilerplate
 
-trait CommonTestData { this: ScalaFutures =>
+trait CommonTestData extends TestComponent with GcsPathUtils{ this: ScalaFutures =>
```
Comment (rtitle) [inline src/test/scala/org/broadinstitute/dsde/workbench/leonardo/CommonTestData.scala]: I am not sure `CommonTestData` should depend on `TestComponent`. I think those concerns should be separated: e.g. it should be possible for tests to use `CommonTestData` without using a mysql database. 

Simply moving `ClusterExtensions` to `TestComponent` is probably a good solution here.

**Comment by rtitle** [comment]
Looks good, just 1 comment about `CommonTestData` vs `TestComponent`.

File: src/test/scala/org/broadinstitute/dsde/workbench/leonardo/CommonTestData.scala (hunk 1 above)
Comment (rtitle) [reply src/test/scala/org/broadinstitute/dsde/workbench/leonardo/CommonTestData.scala]: Hm, I think it'd be better for them to not depend on one another -- it's more modular that way. Ok to redefine `serviceAccountKey` if that's the only thing needed.

**Pull request merged**